File: lib/models/artwork.ts
export interface ArtworkThumbnail {
  lqip: string;
  width: number;
  height: number;
  alt_text?: string;
}

export interface ArtworkDimensionDetail {
  depth?: number | null;
  width?: number | null;
  height?: number | null;
  diameter?: number | null;
  clarification?: string | null;
}

export interface ArtworkColor {
  h: number;
  l: number;
  s: number;
  percentage: number;
  population: number;
}

export interface ArtworkSuggestAutocomplete {
  input: string[];
  weight?: number;
  contexts: {
    groupings: string[];
  };
}

export interface Artwork {
  id: number;
  last_fully_fetched_at: string; // for manual cache management
  image_url?: string; //actual image url
  api_model: string;
  api_link: string;
  is_boosted: boolean;
  title: string;
  alt_titles: string[] | null;
  thumbnail: ArtworkThumbnail;
  main_reference_number: string;
  has_not_been_viewed_much: boolean;
  boost_rank: number | null;
  date_start: number;
  date_end: number;
  date_display: string;
  date_qualifier_title: string;
  date_qualifier_id: number | null;
  artist_display: string;
  place_of_origin: string;
  description: string | null;
  short_description: string | null;
  dimensions: string | null;
  dimensions_detail: ArtworkDimensionDetail[];
  medium_display: string;
  inscriptions: string | null;
  credit_line: string;
  catalogue_display: string | null;
  publication_history: string | null;
  exhibition_history: string | null;
  provenance_text: string | null;
  edition: string | null;
  publishing_verification_level: string;
  internal_department_id: number;
  fiscal_year: number;
  fiscal_year_deaccession: number | null;
  is_public_domain: boolean;
  is_zoomable: boolean;
  max_zoom_window_size: number;
  copyright_notice: string | null;
  has_multimedia_resources: boolean;
  has_educational_resources: boolean;
  has_advanced_imaging: boolean;
  colorfulness: number;
  color: ArtworkColor;
  latitude: number | null;
  longitude: number | null;
  latlon: string | null;
  is_on_view: boolean;
  on_loan_display: string | null;
  gallery_title: string | null;
  gallery_id: number | null;
  nomisma_id: string | null;
  artwork_type_title: string;
  artwork_type_id: number;
  department_title: string;
  department_id: string;
  artist_id: number | null;
  artist_title: string | null;
  alt_artist_ids: number[];
  artist_ids: number[];
  artist_titles: string[];
  category_ids: string[];
  category_titles: string[];
  term_titles: string[];
  style_id: string | null;
  style_title: string | null;
  alt_style_ids: string[];
  style_ids: string[];
  style_titles: string[];
  classification_id: string;
  classification_title: string;
  alt_classification_ids: string[];
  classification_ids: string[];
  classification_titles: string[];
  subject_id: string | null;
  alt_subject_ids: string[];
  subject_ids: string[];
  subject_titles: string[];
  material_id: string | null;
  alt_material_ids: string[];
  material_ids: string[];
  material_titles: string[];
  technique_id: string | null;
  alt_technique_ids: string[];
  technique_ids: string[];
  technique_titles: string[];
  theme_titles: string[];
  image_id: string;
  alt_image_ids: string[];
  document_ids: string[];
  sound_ids: string[];
  video_ids: string[];
  text_ids: string[];
  section_ids: string[];
  section_titles: string[];
  site_ids: string[];
  suggest_autocomplete_all: ArtworkSuggestAutocomplete[];
  source_updated_at: string;
  updated_at: string;
  timestamp: string;
}

export interface ArtworkListApiResponse {
  pagination: {
    total: number;
    limit: number;
    offset: number;
    total_pages: number;
    current_page: number;
    next_url?: string;
  };
  data: Artwork[];
  info: {
    license_text: string;
    license_links: string[];
    version: string;
  };
  config: {
    iiif_url: string;
    website_url: string;
  };
}

export interface ArtworkDetailApiResponse {
  data: Artwork;
  info: {
    license_text: string;
    license_links: string[];
    version: string;
  };
  config: {
    iiif_url: string;
    website_url: string;
  };
}

File: ArtworkApi/ArtworkApi/Models/Artwork.cs
using System;
using System.Collections.Generic;

namespace ArtworkApi.Models
{
    public class Artwork
    {
        public int Id { get; set; }
        public string ApiModel { get; set; }
        public string ApiLink { get; set; }
        public bool IsBoosted { get; set; }
        public string Title { get; set; }
        public List<string> AltTitles { get; set; }
        public object Thumbnail { get; set; } // Could be a more specific type
        public string MainReferenceNumber { get; set; }
        public bool HasNotBeenViewedMuch { get; set; }
        public double? BoostRank { get; set; }
        public int? DateStart { get; set; }
        public int? DateEnd { get; set; }
        public string DateDisplay { get; set; }
        public string DateQualifierTitle { get; set; }
        public int? DateQualifierId { get; set; }
        public string ArtistDisplay { get; set; }
        public string PlaceOfOrigin { get; set; }
        public string Description { get; set; }
        public string ShortDescription { get; set; }
        public string Dimensions { get; set; }
        public object DimensionsDetail { get; set; } // Could be a more specific type
        public string MediumDisplay { get; set; }
        public string Inscriptions { get; set; }
        public string CreditLine { get; set; }
        public string CatalogueDisplay { get; set; }
        public string PublicationHistory { get; set; }
        public string ExhibitionHistory { get; set; }
        public string ProvenanceText { get; set; }
        public string Edition { get; set; }
        public string PublishingVerificationLevel { get; set; }
        public double? InternalDepartmentId { get; set; }
        public double? FiscalYear { get; set; }
        public double? FiscalYearDeaccession { get; set; }
        public bool IsPublicDomain { get; set; }
        public bool IsZoomable { get; set; }
        public double? MaxZoomWindowSize { get; set; }
        public string CopyrightNotice { get; set; }
        public bool HasMultimediaResources { get; set; }
        public bool HasEducationalResources { get; set; }
        public bool HasAdvancedImaging { get; set; }
        public float? Colorfulness { get; set; }
        public object Color { get; set; } // Could be a more specific type
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public string Latlon { get; set; }
        public bool IsOnView { get; set; }
        public string OnLoanDisplay { get; set; }
        public string GalleryTitle { get; set; }
        public double? GalleryId { get; set; }
        public string NomismaId { get; set; }
        public string ArtworkTypeTitle { get; set; }
        public double? ArtworkTypeId { get; set; }
        public string DepartmentTitle { get; set; }
        public double? DepartmentId { get; set; }
        public int? ArtistId { get; set; }
        public string ArtistTitle { get; set; }
        public List<int> AltArtistIds { get; set; }
        public int? ArtistIds { get; set; }
        public List<string> ArtistTitles { get; set; }
        public List<string> CategoryIds { get; set; }
        public List<string> CategoryTitles { get; set; }
        public List<string> TermTitles { get; set; }
        public string StyleId { get; set; }
        public string StyleTitle { get; set; }
        public List<string> AltStyleIds { get; set; }
        public List<string> StyleIds { get; set; }
        public List<string> StyleTitles { get; set; }
        public string ClassificationId { get; set; }
        public string ClassificationTitle { get; set; }
        public List<string> AltClassificationIds { get; set; }
        public List<string> ClassificationIds { get; set; }
        public List<string> ClassificationTitles { get; set; }
        public string SubjectId { get; set; }
        public List<string> AltSubjectIds { get; set; }
        public List<string> SubjectIds { get; set; }
        public List<string> SubjectTitles { get; set; }
        public string MaterialId { get; set; }
        public List<string> AltMaterialIds { get; set; }
        public List<string> MaterialIds { get; set; }
        public List<string> MaterialTitles { get; set; }
        public string TechniqueId { get; set; }
        public List<string> AltTechniqueIds { get; set; }
        public List<string> TechniqueIds { get; set; }
        public List<string> TechniqueTitles { get; set; }
        public List<string> ThemeTitles { get; set; }
        public Guid? ImageId { get; set; }
        public List<string> AltImageIds { get; set; }
        public List<string> DocumentIds { get; set; }
        public Guid? SoundIds { get; set; }
        public Guid? VideoIds { get; set; }
        public Guid? TextIds { get; set; }
        public List<long> SectionIds { get; set; }
        public List<string> SectionTitles { get; set; }
        public List<int> SiteIds { get; set; }
        public object SuggestAutocompleteBoosted { get; set; }
        public object SuggestAutocompleteAll { get; set; }
        public DateTime? SourceUpdatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? Timestamp { get; set; }
    }
}